llsd/serde_notation.py

```diff
@@ -346,21 +346,16 @@ class LLSDNotationFormatter(LLSDBaseFormatter):
 
     See http://wiki.secondlife.com/wiki/LLSD#Notation_Serialization
     """
-    __slots__ = []
-
     def LLSD(self, v):
         return self._generate(v.thing)
     def UNDEF(self, v):
-        return b'!'
+        self.stream.write(b'!')
     def BOOLEAN(self, v):
-        if v:
-            return b'true'
-        else:
-            return b'false'
+        self.stream.write(b'true' if v else b'false')
```

Comment:
Naive Q since I'm probably missing a lot of context.  These were publicish methods that someone might use to generate fragments of notation format for a value before.  Did nobody actually use them for such a purpose because this would break every such use?  Should these all be renamed "_<skfdj>" (they're modifying an internal object)?

Reply:
> These were publicish methods that someone might use to generate fragments of notation format for a value before. Did nobody actually use them for such a purpose because this would break every such use? Should these all be renamed "_" (they're modifying an internal object)?

I have always assumed that the historical names of these methods were chosen to emphasize the communication between [LLSDBaseFormatter](https://github.com/secondlife/python-llsd/blob/main/llsd/base.py#L318-L347) and its subclasses. I've never seen any `llsd` consumer directly reference any `LLSDMumbleFormatter` class, never mind call its type-specific methods.

That said, yes, those methods are intended solely for internal use by that class hierarchy. Perhaps they should be prefixed with an underscore. If we're going to break hypothetical consumers, better to break them overtly with `AttributeError` than by quietly returning `None`.